src/pudl/transform/ferc1.py

```diff
@@ -249,6 +250,56 @@ def align_row_numbers_dbf(
     return df
 
 
+class SelectDateRangeDurationXbrl(TransformParams):
+    """Parameters for selecting date ranges for report_years in the duration table."""
+
+    date_ranges: list[
+        dict[Literal["report_year", "start_date", "end_date"], int | datetime.date]
+    ] = []
+    """List of date ranges.
+
+    Each element of the list must be a dictionary with three keys: report_year,
+    start_date and end_date. The report_year should be an year represented as an int.
+    This will be the report year that you want to apply this date range to. The start
+    and end date must be dates.
+    """
+
+
+def select_date_range_duration_xbrl(
+    df: pd.DataFrame, params: SelectDateRangeDurationXbrl | None = None
+) -> pd.DataFrame:
+    """Select rows that have specific date ranges for each report_year.
+
+    In the XBRL tables, there are occassionally multiple entries for each report year.
+    Sometimes there are sneaky, sneaky quarterly filings that shouldn't be in the annual
+    FERC1 reporting tables. Sometimes this includes data from the previous year. This
+    function enables us to choose specific start_date and end_date for any specific
+    report_year.
+    """
+    if not params:
+        params = SelectDateRangeDurationXbrl()
+    if params.date_ranges:
+        len_og = len(df)
```

Comment:
Please use `orig` rather than `og` to indicate "original".

Reply:
[oh come on](https://www.merriam-webster.com/dictionary/OG)

Reply:
All I can think of is [The Secret World of Og](https://en.wikipedia.org/wiki/The_Secret_World_of_Og_(film))

Reply:
this is the whitest you've ever been.